lib/forms/overlay.vue

```diff
@@ -194,25 +92,66 @@
         }
       },
       hasSections: (state) => state.ui.currentForm.schema.sections && state.ui.currentForm.schema.sections.length > 1,
+      expandedInput() {
+        return expand(this.schema[fieldProp]);
+      },
+      inputName() {
+        return this.expandedInput[inputProp];
+      },
       sections() {
         const currentForm = _.get(this.$store, 'state.ui.currentForm'),
-          sections = _.get(currentForm, 'schema.sections'),
+          schemaSections = _.get(currentForm, 'schema.sections'),
           fields = _.get(currentForm, 'schema.fields'),
           path = _.get(currentForm, 'path'),
-          schema = this.schema;
-
-        if (!_.isEmpty(sections)) {
-          return _.map(sections, (section) => ({
-            title: section.title,
-            fields: section.fields,
-            hasRequiredFields: _.some(schema, (val, key) => _.includes(section.fields, key) && _.has(val, `${fieldProp}.validate.required`))
-          }));
+          currentURI = _.get(this.$store, 'state.ui.currentSelection.uri') || _.get(this.$store, 'state.ui.currentForm.uri'),
+          componentName = getComponentName(currentURI),
+          schema = this.$store.state.schemas[componentName];
+
+
+        let sections = [];
+
+        if (!_.isEmpty(schemaSections)) {
+          sections = _.map(schemaSections, (section) => {
+            const sectionFields = {};
+
+            section.fields.forEach((field) => {
+              sectionFields[field] = {
+                name: field,
+                data: this.fields[field],
+                schema: schema[field],
+                initialFocus: _.get(this.$store, 'state.ui.currentForm.initialFocus')
```

Comment:
could we have this `_.get(this.$store, 'state.ui.currentForm.initialFocus')` outside the `forEach`? Maybe to avoid getting it every time

Reply:
👍 good catch